src/main/java/org/logstash/dissect/DelimZero.java

```diff
@@ -0,0 +1,36 @@
+package org.logstash.dissect;
+
+class DelimZero implements Delim {
```

Comment:
I wonder whether all these specializations could be folded into one (based on `DelimMany` that is just parameterized and you provide a factory method to create the specific variations.